FirebaseRemoteConfig/Sources/RCNConfigContent.m

```diff
@@ -399,11 +399,12 @@ - (BOOL)checkAndWaitForInitialDatabaseLoad {
 }
 
 // Compare fetched config with active config and output what has changed
-- (FIRRemoteConfigUpdate *)getConfigUpdateForNamespace:(NSString *)FIRNamespace {
-  // TODO: handle diff in experiment metadata
-
+- (FIRRemoteConfigUpdate *)getConfigUpdateForNamespace:(NSString *)FIRNamespace
+                                 withExperimentChanges:
+                                     (NSMutableSet<NSString *> *)changedExperimentKeys {
```

Comment:
instead of doing the diffing elsewhere and passing in the changed keys, could we not read the experiment metadata directly from the dbManager here and do the diffing in this method?

feels unnecessary to have to explicitly pass in the experiment-related keys to `getConfigUpdateForNamespace`, when the diffing logic can be self-contained within `RCNConfigContent`

Reply:
My thinking is that all experiment logic lives in RCNConfigExperiments including in-memory copies of active and fetched experiments so why not do the diffing where all of the experiments already are so that we don't need to load all of the data elsewhere to do the same thing. Plus where it's called in Fetch allows us to easily pass in the keys after getting the diffs from experiments

Reply:
But I could go either way